nodestream/model/relationship_draft.py

```diff
@@ -0,0 +1,43 @@
+from dataclasses import dataclass
+from logging import getLogger
+from typing import Optional
+
+from .creation_rules import NodeCreationRule, RelationshipCreationRule
+from .graph_objects import Node, Relationship, RelationshipWithNodes
+
+LOGGER = getLogger(__name__)
+
+
+@dataclass(slots=True)
+class RelationshipDraft:
```

Comment:
Do we need this class? Maybe we can just add unfinished `RelationshipWithNodes` to a separate list and then the finalize method simply adds the to_node and the from_node data?

Reply:
I've been writing some rust recently and they love this [type state](https://cliffle.com/blog/rust-typestate/) pattern. I think this is what was in my head when we were talking about this. Mostly I think this criteria applies here:

> Operations on an object (such as methods or functions) that are only available when the object is in certain states

Now arguably that pattern doesn't really map the same way in python. But I do prefer this separation of methods regardless.